miprometheus/helpers/problem_initializer.py

```diff
@@ -58,54 +62,73 @@ def __init__(self, config=None, name=None, path=None):
 		# If config is provided, parse and build problem from it.
 		else:
 			try:
-				#self.params.add_config_params_from_yaml(os.path.expanduser(config))
+				# self.params.add_config_params_from_yaml(os.path.expanduser(config))
 				configs_to_load = self.recurrent_config_parse(os.path.expanduser(config),[])
 				self.recurrent_config_load(configs_to_load)
 				self.params = self.params['training']
 			except FileNotFoundError:
-				print("Config file at path '{}' not found.".format(config))
+				self.logger.error("Config file at path '{}' not found.".format(config))
 				exit(1)
 
 		# If path is provided, override default path.
 		if path is not None:
-			self.params.add_config_params({'problem': {'data_folder': path}})
+			self.params.add_default_params({'problem': {'data_folder': path}})
 
 		# Pass initialization only flag.
-		self.params.add_config_params({'problem':{'initialization_only': True}})
+		self.params.add_default_params({'problem': {'initialization_only': True}})
 
 		# Build Problem
 		try:
 			_ = ProblemFactory.build(self.params['problem'])
 		except AttributeError:
-			print("Provided problem name not found.")
+			self.logger.error("Provided problem name not found.")
 			exit(1)
 
-# Useful function to properly parse argparse
-def int_or_str(val):
-	try:
-		return int(val)
-	except ValueError:
-		return val
-
-# Useful function to properly parse argparse
-def str_to_bool(val):
-	if val.lower() in ('yes', 'true', 't', 'y', '1'):
-		return True
-	elif val.lower() in ('no', 'false', 'f', 'n', '0'):
-		return False
-	else:
-		raise argparse.ArgumentTypeError('Boolean value expected.')
+	# Useful function to properly parse argparse
+	@staticmethod
```

Comment:
I think those methods should be moved to Worker class, due to their potential utility... but that can wait.